client-core/src/service/wallet_service.rs

```diff
@@ -255,7 +248,7 @@ where
     /// Adds a multi-sig address to given wallet
     pub fn add_root_hash(&self, name: &str, enckey: &SecKey, root_hash: H256) -> Result<()> {
         self.modify_wallet(name, enckey, move |wallet| {
-            wallet.root_hashes.insert(root_hash);
+            wallet.root_hashes.push(root_hash);
```

Comment:
We now need to check here if given `root_hash` already exists and don't push if it does.